Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -1557,7 +1557,11 @@ std::optional<unsigned> gstGetAutoplugSelectResult(ASCIILiteral nick)
 #if !GST_CHECK_VERSION(1, 20, 0)
 GstBuffer* gst_buffer_new_memdup(gconstpointer data, gsize size)
 {
+#if GLIB_CHECK_VERSION(2, 68, 0)
     gpointer copiedData = g_memdup2(data, size);
+#else
+    gpointer copiedData = g_memdup(data, static_cast<unsigned>(size));
+#endif
 
     return gst_buffer_new_wrapped_full(static_cast<GstMemoryFlags>(0), copiedData, size, 0, size, copiedData, g_free);
```

Comment:
The cast would need to be here:

```suggestion
    return gst_buffer_new_wrapped_full(static_cast<GstMemoryFlags>(0), copiedData, size, 0, static_cast<unsigned>(size), copiedData, g_free);
```

Though of course you don't want to do that in the case where you do have `g_memdup2()`.

Yet another option: you could copy the entire function:

```
gpointer
g_memdup2 (gconstpointer mem,
           gsize         byte_size)
{
  gpointer new_mem;

  if (mem && byte_size != 0)
    {
      new_mem = g_malloc (byte_size);
      memcpy (new_mem, mem, byte_size);
    }
  else
    new_mem = NULL;

  return new_mem;
}
```

Reply:
> The cast would need to be here:

Ah, ok, I get what you mean. The cast that I added is not needed because there's already an implicit cast to guint happening. But when calling g_memdup we may get an amount of memory smaller than it was requested because of the cast. This means that the behavior won't be exactly the same in both cases. So I think the proper fix is copying the function, as you mention.